metadata/src/main/java/org/apache/kafka/image/loader/LogDeltaManifest.java

```diff
@@ -27,6 +27,59 @@
  * Contains information about a set of changes that were loaded from the metadata log.
  */
 public class LogDeltaManifest implements LoaderManifest {
+
+    public static class Builder {
+        private MetadataProvenance provenance;
+        private LeaderAndEpoch leaderAndEpoch;
+        private Integer numBatches;
```

Comment:
seems a bit weird to use boxed primitives. It's quite inefficient in Java.

If you want them to start with invalid values, just make them negative (that works for numBatches, elapsedNs, numBytes ... negative values for those don't make sense.)